sonic_platform_base/component_base.py

```diff
@@ -103,3 +103,28 @@ def update_firmware(self, image_path):
             RuntimeError: update failed
         """
         raise NotImplementedError
+
+    def auto_update_firmware(self, image_path, boot_action):
```

Comment:
@sujinmkang since this API is a part of SONiC firmware upgrade framework, please provide the description on which boot types are supported